### ISSUE TITLE
map(fn, iterable, ...) should be able to take in more than 2 iterables

### ISSUE THREAD

**Issue by yi-lu** [python/typeshed#1979]
The built-in `map()` function should be able to accept a list of iterables, however only one and two args are special cased by typeshed: https://github.com/python/typeshed/blob/master/stdlib/2/__builtin__.pyi#L755
If trying to pass in 3 iterables: 
```
def fn(a, b, c):
# type: (int, int, int) -> int
return a+b+c

def map_fn(v):
# type: (int) -> Sequence[int]
return map(fn, [v]*3, [v]*3, [v]*3)
```
it'll throw
```
error: No overload variant of "map" matches argument types [def (a: builtins.int, b: builtins.int, c: builtins.int) -> builtins.int, builtins.list[builtins.int*], builtins.list[builtins.int*], builtins.list[builtins.int*]]
```

**Comment by gvanrossum** [comment]
Sure, do you want to submit a PR? E.g. for `zip()` we go up to 5 and we also have a version with 6 or more args and a less precise return type.

**Comment by sproshev** [comment]
https://github.com/python/typeshed/pull/1990

**Comment by gvanrossum** [comment]
Fixed by #1990.
